independent-projects/arc/runtime/src/main/java/io/quarkus/arc/InstanceImpl.java

```diff
@@ -80,7 +81,13 @@ public T get() {
         } else if (beans.size() > 1) {
             throw new AmbiguousResolutionException("Beans: " + beans.toString());
         }
-        return getBeanInstance((InjectableBean<T>) beans.iterator().next());
+        InjectableBean<T> bean = (InjectableBean<T>) beans.iterator().next();
```

Comment:
This is not correct. `IllegalProductException` should only be thrown for producer beans with a normal scope. This would throw an IPE even for `@Dependent` producers that are allowed to return `null`.